### ISSUE TITLE
Plugin crashes with wrong credentials

### ISSUE THREAD

**Issue by mandrav** [ddelnano/terraform-provider-mikrotik#63]
Hi,

the plugin would always crash on me until I realized I had changed my router and did not re-create the automation user!
I had a look at the code and it seems you 're not checking that the `client` variable is valid. For example, here: https://github.com/ddelnano/terraform-provider-mikrotik/blob/c2a1c2c1e254957f8fb92e328214be2212ee7088/mikrotik/resource_dns_record.go#L48

In the above code, `c` is nil when the credentials are wrong, which leads to this stack trace:

```
Stack trace from the terraform-provider-mikrotik_v0.9.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x626307]

goroutine 70 [running]:
github.com/go-routeros/routeros.(*Client).RunArgs(0x0, {0xc000494600, 0x2, 0xc0004ad790})
        github.com/go-routeros/routeros@v0.0.0-20210123142807-2a44d57c6730/run.go:21 +0x27
github.com/ddelnano/terraform-provider-mikrotik/client.Mikrotik.FindDnsRecord({{0xc0004ad4a0, 0x10}, {0xc0004ad4d0, 0x9}, {0xc0004716f8, 0x18}, 0x0, {0x0, 0x0}, 0x0, ...}, ...)
        github.com/ddelnano/terraform-provider-mikrotik/client@v0.0.0-00010101000000-000000000000/dns.go:42 +0x10b
github.com/ddelnano/terraform-provider-mikrotik/mikrotik.resourceServerRead({0xc11eb8, 0xc0000bd5c0}, 0xc000159080, {0xb176e0, 0xc000307500})
        github.com/ddelnano/terraform-provider-mikrotik/mikrotik/resource_dns_record.go:61 +0x165
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0003ae000, {0xc11eb8, 0xc0000bd5c0}, 0x24, {0xb176e0, 0xc000307500})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:358 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0003ae000, {0xc11eb8, 0xc0000bd5c0}, 0xc0003404e0, {0xb176e0, 0xc000307500})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0000b40a8, {0xc11eb8, 0xc0000bd5c0}, 0xc0000bd600)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0000baca0, {0xc11f60, 0xc0000f45a0}, 0xc000307860)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:298 +0x1fb
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xaf8a40, 0xc0000baca0}, {0xc11f60, 0xc0000f45a0}, 0xc000307800, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000dc700, {0xc1db38, 0xc000404300}, 0xc0000c0800, 0xc000400630, 0x108bc70, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc0000dc700, {0xc1db38, 0xc000404300}, 0xc0000c0800, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x294

Error: The terraform-provider-mikrotik_v0.9.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Anyway, great job with the plugin!

**Comment by maxim-nazarenko** [comment]
@ddelnano if you don't mind, I would like to pick this one up.

**Comment by maxim-nazarenko** [comment]
Hi @mandrav , the issue should be resolved in #66 .

**Comment by ddelnano** [comment]
The fix will be included in v0.9.1.
